### PR TITLE
add types for `math-toolbox`

### PR DESCRIPTION
**Pull request by devloop01**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/math-toolbox/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,582 @@
+/**
+ * Constrain a value to lie between two further values.
+ *
+ * @param  {number} min - Lower end of the range into which to constrain v.
```
Comment (MysteryBlokHed) [inline types/math-toolbox/index.d.ts:4]: JSDoc type hints like `{number}` aren't needed in TypeScript files

Comment (devloop01) [reply types/math-toolbox/index.d.ts:4]: oh! sorry
The file was autogenerated from jsdoc comments I just added them here 
Should I remove them?

Comment (MysteryBlokHed) [reply types/math-toolbox/index.d.ts:4]: I'd recommend it, since it can be confusing with Typescript's types already there

Comment (devloop01) [reply types/math-toolbox/index.d.ts:4]: got it thanks!
will fix it tomorrow gotta sleep now 😴

**Review by jakebailey — CHANGES_REQUESTED** [review]
After #67332, this PR will need to be updated to include an `.npmignore`. See https://github.com/DefinitelyTyped/DefinitelyTyped#npmignore

File: types/math-toolbox/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+ * @param  a - The start value.
+ * @param  b - The end value.
+ * @param  t - Value to inerpolate.
+ * @return The interpolated value.
+ * @see {@link https://docs.unity3d.com/ScriptReference/Mathf.LerpAngle.html}
+ */
+declare function lerpAngleDeg(a: number, b: number, t: number): number;
+
+export {
+    almostIdentity,
```
Comment (jakebailey) [inline types/math-toolbox/index.d.ts:523]: Why the big export block instead of explicitly exporting each?

Comment (devloop01) [reply types/math-toolbox/index.d.ts:523]: umm no reason just how it was generated from jsdoc comments. Should I fix this?

Comment (jakebailey) [reply types/math-toolbox/index.d.ts:523]: It seems like it'd be clearer and easier to modify if they were plain exports, IMO. Additionally, the duplicated exports seem odd; they are not documented in https://terkelg.github.io/math-toolbox/.

Comment (devloop01) [reply types/math-toolbox/index.d.ts:523]: okay will fix the exports and I not get the last part

Comment (jakebailey) [reply types/math-toolbox/index.d.ts:523]: You have:

```
    deltaAngleDeg,
    deltaAngleDeg as deltaAngle,
```

But the docs page doesn't list `deltaAngle`, only `deltaAngleDeg`.

Comment (devloop01) [reply types/math-toolbox/index.d.ts:523]: yes in docs they are not documented but in code there are two exports
(code)[https://github.com/terkelg/math-toolbox/blob/master/src/delta-angle-deg.js]
```js
export { deltaAngleDeg, deltaAngleDeg as deltaAngle }
```

**Review by jakebailey — APPROVED** [review]

**Comment by devloop01** [comment]
Ready to merge

**Pull request merged**